lib/buildingsync/makers/workflow_maker.rb

```diff
@@ -431,52 +431,337 @@ def cleanup_larger_files(osw_dir)
+      native_units.text = fuel_natural_gas_kbtu.to_s
+      consistent_units = REXML::Element.new("#{@ns}:AnnualFuelUseConsistentUnits")
+      consistent_units.text = (fuel_natural_gas_kbtu / 1000.0).to_s # in MMBtu
+      res_use.add_element(energy_res)
+      res_use.add_element(res_units)
+      res_use.add_element(native_units)
+      res_use.add_element(consistent_units)
+      res_uses.add_element(res_use)
+      return res_uses
+    end
+
+    def get_timeseries_data_element(monthly_results,year_val, scenario_name)
+      month_lookup = {1 => 'jan', 2 => 'feb', 3 => 'mar', 4 => 'apr', 5 => 'may', 6 => 'jun', 7 => 'jul', 8 => 'aug', 9 => 'sep', 10 => 'oct', 11 => 'nov', 12 => 'dec'}
+      scenario_name_ns = scenario_name.gsub(' ', '_').gsub(/[^0-9a-z_]/i, '')
+      timeseriesdata = REXML::Element.new("#{@ns}:TimeSeriesData")
+
+      # Electricity
+      # looking for: "electricity_ip_jan" through "electricity_ip_dec"
+      # convert from kWh to kBtu
+      (1..12).each do |month|
+        timeseries = REXML::Element.new("#{@ns}:TimeSeries")
+        reading_type = REXML::Element.new("#{@ns}:ReadingType")
+        reading_type.text = 'Total'
+        timeseries.add_element(reading_type)
+        ts_quantity = REXML::Element.new("#{@ns}:TimeSeriesReadingQuantity")
+        ts_quantity.text = 'Energy'
+        timeseries.add_element(ts_quantity)
+        start_time = REXML::Element.new("#{@ns}:StartTimeStamp")
+        if month < 10
+          start_time.text = year_val.to_s + '-0' + month.to_s + '-01T00:00:00'
+        else
+          start_time.text = year_val.to_s + '-' + month.to_s + '-01T00:00:00'
+        end
+        timeseries.add_element(start_time)
+        end_time = REXML::Element.new("#{@ns}:EndTimeStamp")
+        if month < 9
+          end_time.text = year_val.to_s + '-0' + (month + 1).to_s + '-01T00:00:00'
+        elsif month < 12
+          end_time.text = year_val.to_s + '-' + (month + 1).to_s + '-01T00:00:00'
+        else
+          end_time.text = year_val.to_s + '-01-01T00:00:00'
+        end
+        timeseries.add_element(end_time)
+        interval_frequency = REXML::Element.new("#{@ns}:IntervalFrequency")
+        interval_frequency.text = 'Month'
+        timeseries.add_element(interval_frequency)
+        interval_reading = REXML::Element.new("#{@ns}:IntervalReading")
+        the_key = "electricity_ip_#{month_lookup[month]}"
+        # puts "saving value 123: #{monthly_results[scenario_name][the_key]}"
+        if !monthly_results[scenario_name][the_key.to_sym].nil?
+          interval_reading.text = monthly_results[scenario_name][the_key.to_sym] * 3.4121416331 # kWh to kBtu
+        end
+        timeseries.add_element(interval_reading)
+        resource_id = REXML::Element.new("#{@ns}:ResourceUseID")
+        resource_id.add_attribute('IDref', scenario_name_ns + '_Electricity')
+        timeseries.add_element(resource_id)
+        timeseriesdata.add_element(timeseries)
+      end
```

Comment:
can we refactor so this portion of code related to electricity is using the same function as for natural gas below? Parameterise the function and call it twice ones for electricity and once for gas.